### PR TITLE
Add proceedOriginalCall dsl

### PR DESCRIPTION
**Pull request by ujizin**
Issue: #25 

### Preview

https://github.com/user-attachments/assets/210dbc47-0221-40dc-9bbb-43dad9daf524

### CODE REVIEW

**Comment by ujizin** [comment]
Maestro might be failing due to wrong version of the library, I created another PR resolving #37

File: iris-mock/src/main/kotlin/dev/arildo/iris/mock/dsl/ConditionDsl.kt [hunk 1]
```diff
@@ -126,3 +126,19 @@ fun IrisMockCondition.mockEmptyJsonObject() = "{}"
...
+ *   onGet("/products") then {
+ *     if (!BuildConfig.DEBUG) {
+ *       proceedOriginalCall()
+ *     }
+ *   }
+ * }
+ * */
+fun IrisMockCondition.proceedOriginalCall() {
+    irisMockScope.proceedOriginalResponse()
+}
```
Comment (arildojr7) [inline iris-mock/src/main/kotlin/dev/arildo/iris/mock/dsl/ConditionDsl.kt:144]: missing end line

**Comment by arildojr7** [comment]
instead of adding the `useOriginalResponse`, you could just change the value of `useCustomResponse`, so the function would looks like:
```kotlin
fun IrisMockCondition.proceedOriginalCall() {
    irisMockScope.useCustomResponse = false
}
```

**Comment by ujizin** [comment]
Sorry for the late response! Since there were some refactors on main branch, I believe this will need to be remake or not be necessary anymore, I could proceed with original call doing this:

```Kotlin
override fun intercept(chain: Interceptor.Chain) = irisMock(chain) {
  onGet(endsWith = "/public/characters") {
    if (BuildConfig.DEBUG) {
      mockResponse("") // modify if necessary
    }
    // proceed with original call, because nothing was modified
  }
}
```
I will close this PR but if necessary I will open another one 😄 

and thanks for the awesome library!

**Pull request closed without merging**